### PR TITLE
Improve tdbg command

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Use frontend delete workflow API, not admin API which target database directly
* Always bypass API redirection

**Why?**
Make tdbg useful in XDC env

**How did you test it?**
N/A

**Potential risks**
N/A

**Is hotfix candidate?**
N/A

### CODE REVIEW

**Review by yux0 — APPROVED** [review]

File: tools/tdbg/commands.go [hunk 1]
```diff
@@ -222,25 +223,16 @@ func AdminDeleteWorkflow(c *cli.Context) error {
 	ctx, cancel := newContext(c)
 	defer cancel()
 
-	resp, err := adminClient.DeleteWorkflowExecution(ctx, &adminservice.DeleteWorkflowExecutionRequest{
+	_, err = frontendClient.DeleteWorkflowExecution(ctx, &workflowservice.DeleteWorkflowExecutionRequest{
```
Comment (yycptt) [inline tools/tdbg/commands.go:226]: the DeleteWorkflowExecution() api in frontend client is not force delete, so it won't work for corrupted workflows.

Comment (yycptt) [reply tools/tdbg/commands.go:226]: for normal workflow deletion, plz use the delete workflow execution command in temporal cli.

**Comment by yycptt** [comment]
Would you mind provide some context re. ?
> Make tdbg useful in XDC env

My understanding is tdbg is not supposed to be used by service code directly, but for cluster admin to do troubleshooting and incident mitigation.

File: tools/tdbg/commands.go (hunk 1 above)
Comment (wxing1292) [reply tools/tdbg/commands.go:226]: >the DeleteWorkflowExecution() api in frontend client is not force delete, so it won't work for corrupted workflows for troubleshooting purpose.

yes but delete API directly use database layer API

**Comment by wxing1292** [comment]
> Would you mind provide some context re. ?
> 
> > Make tdbg useful in XDC env
> 
> My understanding is tdbg is not supposed to be used by service code directly, but for cluster admin to do troubleshooting and incident mitigation.

tdbg is an admin tool, so tdbg should operate on local cluster data, hence no API redirection

File: tools/tdbg/commands.go (hunk 1 above)
Comment (yycptt) [reply tools/tdbg/commands.go:226]: Is the concern here frontend service should not access database layer API directly? If so, [this PR](https://github.com/temporalio/temporal/pull/4453) will forward that call to history service for processing.

**Comment by yycptt** [comment]
> > Would you mind provide some context re. ?
> > > Make tdbg useful in XDC env
> > 
> > 
> > My understanding is tdbg is not supposed to be used by service code directly, but for cluster admin to do troubleshooting and incident mitigation.
> 
> tdbg is an admin tool, so tdbg should operate on local cluster data, hence no API redirection

I am confused. Changing tdbg to use frontend client, means the DeleteWorkflowExecution api will be redirected right and that's not what you want? 

Maybe I am still missing something here. To me this PR is making tdbg DeleteWorkflow the same as temporal CLI DeleteWorkflow command, and I am still trying to understand the motivation behind that. Also with this change, it means we no longer have the command for forcefully deleting a workflow.

**Comment by wxing1292** [comment]
> Changing tdbg to use frontend client, means the DeleteWorkflowExecution api will be redirected right and that's not what you want?

the additional ctx will bypass redirection

**Comment by wxing1292** [comment]
> Maybe I am still missing something here. To me this PR is making tdbg DeleteWorkflow the same as temporal CLI DeleteWorkflow command, and I am still trying to understand the motivation behind that. Also with this change, it means we no longer have the command for forcefully deleting a workflow.

when the frontend call execution store to delete a workflow, will frontend call DB directly (# 1) or will frontend call be redirected to history, and history will delete the workflow (# 2)? 

i believe current impl still use #1?

**Comment by yiminc** [comment]
Plan is to forward admin delete request from FE to history. https://github.com/temporalio/temporal/blob/5ab526543a29a26e6a96de050a8ade8c0273272d/service/frontend/admin_handler.go#L1582C16-L1582C29

**Pull request closed without merging**